src/overlay/OverlayManagerImpl.cpp

```diff
@@ -959,11 +966,10 @@ OverlayManagerImpl::shufflePeerList(std::vector<Peer::pointer>& peerList)
 }
 
 bool
-OverlayManagerImpl::recvFloodedMsgID(StellarMessage const& msg,
-                                     Peer::pointer peer, Hash& msgID)
+OverlayManagerImpl::recvFloodedMsg(Peer::pointer peer, Hash const& msgID)
```

Comment:
This change makes it quite error prone: we had bugs in the past because of hash scheme mismatch. While we may not be able to avoid this at the floodmap level, we can avoid exposing this complexity outside of `OverlayManager`.
If we want to track transactions and msgs separately, we should just add  overloads at the `OverlayManager` level that take both (and do the right thing when passing a transaction wrapped in a `StellarMessage`)

Reply:
Yeah I wasn't super happy with this approach either. The problem is that with adverts, we only have transaction hashes, not full messages, so we have to provide an interface that only takes a hash. So I figured the simplest thing would be to force callers to compute the "keys" for whatever messages they want to remember